api/bazel/api_build_system.bzl

```diff
@@ -24,21 +24,39 @@ def _LibrarySuffix(library_name, suffix):
 # https://github.com/bazelbuild/bazel/issues/3935 and/or
 # https://github.com/bazelbuild/bazel/issues/2626 are resolved.
 def api_py_proto_library(name, srcs = [], deps = [], has_services = 0):
-    py_proto_library(
+    _py_proto_library(
         name = _Suffix(name, _PY_SUFFIX),
         srcs = srcs,
         default_runtime = "@com_google_protobuf//:protobuf_python",
         protoc = "@com_google_protobuf//:protoc",
         deps = [_LibrarySuffix(d, _PY_SUFFIX) for d in deps] + [
             "@com_envoyproxy_protoc_gen_validate//validate:validate_py",
-            "@googleapis//:api_httpbody_protos_py",
-            "@googleapis//:http_api_protos_py",
-            "@googleapis//:rpc_status_protos_py",
+            "@com_google_googleapis//google/rpc:status_py_proto",
+            "@com_google_googleapis//google/api:annotations_py_proto",
+            "@com_google_googleapis//google/api:http_py_proto",
+            "@com_google_googleapis//google/api:httpbody_py_proto",
             "@com_github_gogo_protobuf//:gogo_proto_py",
         ],
         visibility = ["//visibility:public"],
     )
 
+# This replaces googleapis rules.
+def py_proto_library(name, deps = []):
+    srcs = [dep[:-6] + ".proto" if dep.endswith("_proto") else dep for dep in deps]
+    proto_deps = []
+
+    # Ignoring the proto_library has an unfortunate side-effect of losing proto dependencies...
```

Comment:
How are we ignoring the proto_library? We're doing some transform in srcs above, can you comment on what the intent there is?

Reply:
`py_proto_library` as defined in googleapis depend on `*_proto`. So we are losing the dependencies of `*_proto` when we refer to the corresponding `.proto` file. Protobuf rules do not work with `*_proto`, so the below is a patch to provide the missing dependencies.

Reply:
OK, more detail in comments please :)

Reply:
Updated.